CONTRIBUTING.rst

```diff
@@ -0,0 +1,268 @@
+   any two different arbitrary string values. Also set ``DB_ENCRYPTION_KEY``::
+
+    $ python
+    >>> from cryptography.fernet import Fernet
+    >>> Fernet.generate_key()
+
+   This will output a bytestring, e.g. ``b'mystring='``. Copy only the contents
+   of ``'...'``, e.g. ``DB_ENCRYPTION_KEY="mystring="``.
+
+   Finally, set the following environment variables (if you're an OddBird, you
+   can find these values in the shared Keybase team folder --
+   ``metashare/env``)::
+
+    BUCKETEER_AWS_ACCESS_KEY_ID=...
+    BUCKETEER_AWS_SECRET_ACCESS_KEY=...
+    BUCKETEER_BUCKET_NAME=...
+
+3. Run ``docker-compose build`` to build/re-build all the container images.
+
+4. Run ``docker-compose up`` to start the server(s).
+
+5. Visit `<http://localhost:8080/>`_ in your browser.
+
+6. When you're done working on MetaShare, ``Ctrl-C`` in the terminal where the
+   containers are running to exit. You can also ``docker-compose down`` to clean
+   up all running containers. (``docker-compose ps`` will tell you what
+   containers are currently running.)
+
+.. _Docker Desktop (Community Edition): https://www.docker.com/products/docker-desktop
+
+Docker development tasks
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To run any development tasks (such as changing Python or JS dependencies, or
+generating or running migrations, or running a Django shell), you will need to
+run them inside the appropriate Docker image. This takes the general form
+``docker-compose run --no-deps [web or client] [command]``. In some cases, such
+as for migrations or a Django shell, you will want to omit the ``--no-deps``
+flag.
+
+You shouldn't need to run any of the setup tasks; the Docker images will take
+care of setting up a database and installing Python and JS dependencies for you.
+
+When you change Python or JS dependencies, you will need to rebuild the Docker
+images, as we store dependencies in the images for speed. This requires running
+``docker-compose build [web or client]``.
+
+Tests and linting can be run similarly::
+
+    $ docker-compose run --no-deps client yarn lintjs
+    $ docker-compose run --no-deps client yarn stylelint
+    $ docker-compose run --no-deps client yarn test
+
+    $ docker-compose run --no-deps web black manage.py metashare/ config/
+    $ docker-compose run --no-deps web isort -rc manage.py metashare/ config/
+    $ docker-compose run --no-deps web flake8 manage.py metashare/ config/
+    $ docker-compose run web pytest
+
+Logging in with Salesforce
+--------------------------
+
+To setup the Salesforce OAuth integration, run the ``populate_social_apps``
+management command. The values to use in place of the ``XXX`` and ``YYY`` flags
+can be found on the Connected App you've made in your Salesforce configuration,
+or if you're an OddBird, you can find these values in the shared Keybase team
+folder (``metashare/prod.db``)::
+
+    $ docker-compose run web python manage.py populate_social_apps --prod-id XXX --prod-secret YYY
```

Comment:
This command fails for me:

```
  File "/app/metashare/multisalesforce/management/commands/populate_social_apps.py", line 37, in handle
    self.create_production_app(options["prod_id"], options["prod_secret"])
  File "/app/metashare/multisalesforce/management/commands/populate_social_apps.py", line 27, in create_production_app
    self._create_app("production", "https://login.salesforce.com/", id, secret)
  File "/app/metashare/multisalesforce/management/commands/populate_social_apps.py", line 21, in _create_app
    name=f"Salesforce {name.title()}", key=key, client_id=id, secret=secret
  File "/usr/local/lib/python3.7/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 538, in get_or_create
    return self.get(**kwargs), False
  File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 402, in get
    num = len(clone)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 256, in __len__
    self._fetch_all()
  File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 1242, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/usr/local/lib/python3.7/site-packages/django/db/models/query.py", line 55, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/usr/local/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 1100, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 99, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.7/site-packages/newrelic/hooks/database_psycopg2.py", line 51, in execute
    **kwargs)
  File "/usr/local/lib/python3.7/site-packages/newrelic/hooks/database_dbapi2.py", line 25, in execute
    *args, **kwargs)
django.db.utils.ProgrammingError: relation "socialaccount_socialapp" does not exist
LINE 1: ...p"."secret", "socialaccount_socialapp"."key" FROM "socialacc...
```

Reply:
Mm, we're lacking some apps I think. I'll get on it.

Reply:
Is it that, or not having run the initial migration?

Reply:
`start-dev.sh`, which the `web` Dockerfile runs, runs migrations.